src/derivative_operators/ghost_derivative_operator.jl

```diff
@@ -1,74 +1,39 @@
-struct GhostDerivativeOperator{T<:Real, E<:AbstractDerivativeOperator{T}, F<:AbstractBC{T}} <: AbstractDiffEqLinearOperator{T}
+struct GhostDerivativeOperator{T, E<:AbstractDiffEqLinearOperator{T}, F<:AbstractBC{T}} <: AbstractDiffEqLinearOperator{T}
     L :: E
     Q :: F
 end
 
-function *(L::AbstractDerivativeOperator{T}, Q::AbstractBC{T}) where{T}
+function *(L::AbstractDiffEqLinearOperator{T}, Q::AbstractBC{T}) where{T}
     return GhostDerivativeOperator{T, typeof(L), typeof(Q)}(L,Q)
 end
 
 function *(L::AbstractDiffEqCompositeOperator{T}, Q::AbstractBC{T}) where{T}
     return sum(map(op -> op * Q, L.ops))
 end
 
-function LinearAlgebra.mul!(x::AbstractVector, A::GhostDerivativeOperator, u::AbstractVector)
-    @assert length(u) == A.L.len == length(x)
+function *(A::GhostDerivativeOperator{T1}, u::AbstractArray{T2}) where {T1,T2}
+    #TODO Implement a function domaincheck(L::AbstractDiffEqLinearOperator, u) to see if components of L along each dimension match the size of u
```

Comment:
that would be nice. Worth opening an issue.

Reply:
I'll open an issue.  I basically wrote the code for this in one of the `mul!' methods, but with a caveat.

Reply:
I've confused myself.  The code I wrote just shaved extraneous padding off padded `u`s for calculation of the derivative in a single direction, and made sure `x` is compatible with some plausible `u` (with the caveat that right now it assumes 1 padding, as the code it replaces did).  I made #280, though.